--- origsrc/ffmpegthumbnailer-2.0.6/Makefile.am	2011-01-06 13:03:03.000000000 -0600
+++ src/ffmpegthumbnailer-2.0.6/Makefile.am	2011-02-27 18:52:00.199873000 -0600
@@ -38,7 +38,7 @@ nobase_include_HEADERS =    libffmpegthu
 
 libffmpegthumbnailer_includedir         = $(includedir)/libffmpegthumbnailer
 libffmpegthumbnailer_la_CXXFLAGS        = $(AM_CXXFLAGS) $(FFMPEG_CFLAGS) $(PNG_CFLAGS)
-libffmpegthumbnailer_la_LDFLAGS         = -version-info 4:7:0
+libffmpegthumbnailer_la_LDFLAGS         = -version-info 4:7:0 -no-undefined
 libffmpegthumbnailer_la_LIBADD          = $(FFMPEG_LIBS) $(PNG_LIBS) $(JPEG_LIBS)
 
 bin_PROGRAMS = ffmpegthumbnailer
