--- origsrc/ffmpegthumbnailer-2.0.6/main.cpp	2011-01-06 13:01:51.000000000 -0600
+++ src/ffmpegthumbnailer-2.0.6/main.cpp	2011-02-27 18:50:14.012888600 -0600
@@ -218,9 +218,15 @@ typedef void (*UnrefFunc)(void*);
 
 void tryUriConvert(std::string& filename)
 {
+#ifdef __CYGWIN__
+    void* gLib = dlopen("cygglib-2.0-0.dll", RTLD_LAZY);
+    void* gobjectLib = dlopen("cyggobject-2.0-0.dll", RTLD_LAZY);
+    void* gioLib = dlopen("cyggio-2.0-0.dll", RTLD_LAZY);
+#else
     void* gLib = dlopen("libglib-2.0.so", RTLD_LAZY);
     void* gobjectLib = dlopen("libgobject-2.0.so", RTLD_LAZY);
     void* gioLib = dlopen("libgio-2.0.so", RTLD_LAZY);
+#endif
 
     if (gioLib && gLib && gobjectLib)
     {   
